src/app/components/Header.jsx

```
@@ -45,6 +45,48 @@ export default function Header() {
           )}
         </div>
       </div>
+      {/* Mobile menu toggle + responsive behavior */}
+{(() => {
+  const [isOpen, setIsOpen] = useState(false);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)